src/utils/hooks/useToggle.tsx

```diff
@@ -0,0 +1,15 @@
+import { useReducer } from 'react';
+
+export type DispatchWithOptionalAction<Type> = (_arg?: Type | unknown) => void;
+
+export type UseToggleType<Type> = [Type, DispatchWithOptionalAction<Type>];
+
+function toggle(currentValue: boolean, newValue: boolean | undefined) {
```

Comment:
It doesn't look right to me. Can you explain the use case here? If we can set the value manually, it shouldn't be called `useToggle` then


The API I expected from the `useToggle` is:
- it takes a boolean as the initial value `useToggle(true|false)`
- it exposes the current value
- it exposes a function to toggle from the current state to the opposite: `toggle()`

Reply:
I guess the idea is to let the user force a state.

To also extend the idea of usage, imagine we have a bunch of expendable components and we want to close them all we can call `toggle(false)` instead of checking each value and if it's true then we close it.
Maybe not the best example but I hope it can be clearer.

I guess the API would be the same with a small correction: 
- it takes a boolean as the initial value `useToggle(true|false)`
- it exposes the current value
- it exposes a function to toggle from the current state to the opposite: `toggle()` || passes the boolean to set `toggle(true)`

Reply:
What do you think about exposing another function to enforce the state? Let's say `setToggle` to make it clear

Reply:
meaning returning something like this?
```js
return [value, toggle, setToggle]
```

Reply:
Yep, or even just set since we are in the context of the toggle